test/sys/test_uio.rs

```diff
@@ -190,3 +190,51 @@ fn test_preadv() {
     let all = buffers.concat();
     assert_eq!(all, expected);
 }
+
+#[test]
+#[cfg(any(target_os = "linux", target_os = "android"))]
+// FIXME: qemu-user doesn't implement process_vm_readv/writev on most arches
+#[cfg_attr(not(any(target_arch = "x86", target_arch = "x86_64")), ignore)]
+fn test_process_vm_readv() {
+    use nix::unistd::ForkResult::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::*;
+    use std::{str, slice};
+
+    let (r, w) = pipe().unwrap();
+    match fork() {
```

Comment:
Forking is bad within the Rust test harness as it can cause deadlocks. Any other way you can implement this test? Also some comments in here explaining things for those unfamiliar with the code would be nice.